js_modules/dagit/src/configeditor/ConfigEditor.tsx

```diff
@@ -23,7 +23,7 @@ import { LintJson as YamlModeLintJson } from "./codemirror-yaml/mode";
 import { debounce } from "../Util";
 
 interface IConfigEditorProps {
-  pipeline: ConfigEditorPipelineFragment;
+  pipeline: ConfigEditorPipelineFragment | "loading";
```

Comment:
yeah I wasn't sure what we should do here. These components can be displayed without a `pipeline` (which is preferable to not displaying them at all until the pipeline has been fetched) but don't become interactable / fully usable until the pipeline is loaded. I didn't want to make the prop optional since it might seem like using the component without a pipeline was a valid use case. I figured `pipeline | "loading"` made the two available states more clear than `pipeline | undefined`.

I'll switch 'em to be optional - I think that is more standard.